### DISCUSSION TITLE
Extracting motion parameters in registration

### DISCUSSION THREAD

**Discussion by OscarWC** [Project-MONAI/MONAI#6128] [Q&A]
When completing registration, rather than warping the moving image to get a predicted image, is there a way to get the motion parameters for affine transformations between moving and fixed images through GlobalNet or any other network. Thank you in advance for the help.

**Comment by wyli** [comment]
I think you can follow this approach in general https://discuss.pytorch.org/t/how-can-i-extract-intermediate-layer-output-from-loaded-cnn-model/77301

for globalnet:
```py
import torch
from monai.networks.blocks import Warp
from monai.networks.nets import GlobalNet

activation = {}


def get_activation(name):
    def hook(model, input, output):
        activation[name] = output.detach()

    return hook


input_param = {
    "image_size": (16, 16),
    "spatial_dims": 2,
    "in_channels": 1,
    "num_channel_initial": 16,
    "depth": 1,
    "out_kernel_initializer": "kaiming_uniform",
    "out_activation": None,
    "pooling": True,
    "concat_skip": True,
    "encode_kernel_sizes": 3,
}
net = GlobalNet(**input_param)
net.output_block.fc.register_forward_hook(get_activation("fc"))
warp_layer = Warp()
img = torch.randn((1, 1, 16, 16))
result = net(img)
warped = warp_layer(img, result)

print(activation)
```

**Reply by OscarWC** [reply]
Thank you very much for this, it has given me results using my own data. I was hoping you could explain to me what the numbers correspond to for each 6-parameter tensor I get back for each image. Many thanks.

**Reply by wyli** [reply]
sure, these parameters are used by the affine transform head: https://github.com/Project-MONAI/MONAI/blob/1b0808db48b2a3c0eb4287c6290bd754e70fb88b/monai/networks/nets/regunet.py#L236-L244

**Reply by OscarWC** [reply]
@wyli apologies I am still a bit confused on what the output as 'out_init' actually means for each of the 6 output parameters for a 2D case (or even for the 12 parameters in a 3D case).

**Comment by wyli** [comment]
now it's also addressed by https://github.com/Project-MONAI/MONAI/pull/6459
